src/service/FileTransferService.ts

```diff
@@ -42,10 +46,75 @@ export class FileTransferService {
                 }
             }).catch((err) => {
                 if (err.code === UNSUPPORTED_MEDIA_TYPE) {
-                    throw new Error('Unsupported file type')
+                    throw new Error('Unsupported file type');
                 } else {
-                    throw new Error(err.message)
+                    throw new Error(err.message);
                 }
             });
     }
+
+    async getFileMetadata(fileId: string): Promise<FileMetadata> {
+
+        const config: AxiosRequestConfig = {
+            headers: {
+                'x-api-key': this.key
+            },
+        };
+
+        return await axios
+            .get<FileMetadata>(`${this.url}/${fileId}`, config)
+            .then((axiosResponse: AxiosResponse<FileMetadata>) => axiosResponse.data)
+            .catch(err => {
+                throw this.getErrorFrom(err, fileId);
+            });
+    }
+
+    async download(fileId: string,
+        writableStream: stream.Writable,
+        onStart?: (axiosResponse: AxiosResponse<stream.Readable>) => void): Promise<void> {
+
+        const url = `${this.url}/${fileId}/download`;
+        const config: AxiosRequestConfig = {
+            headers: {
+                'x-api-key': this.key
+            },
+            responseType: 'stream'
+        };
+
+        return axios.get<stream.Readable>(url, config)
+            .then(async (axiosResponse: AxiosResponse<stream.Readable>) => {
+                if (onStart) {
+                    onStart(axiosResponse);
+                }
+                await this.pipeDataIntoStream(axiosResponse, writableStream);
+            })
+            .catch(err => {
+                throw this.getErrorFrom(err, fileId);
+            });
+
+    }
+
+    public async pipeDataIntoStream(axiosResponse: AxiosResponse<stream.Readable>,
+        writableStream: stream.Writable): Promise<void> {
+
+        return new Promise<void>((resolve, reject) => axiosResponse.data.pipe(writableStream)
+            .on('finish', resolve)
+            .on('error', reject));
+    }
+
+    private getErrorFrom(err: any, fileId: string): Error {
+
+        if (err.isAxiosError) {
+            switch (err.response.status) {
+                case NOT_FOUND:
+                    return new FileNotFoundError(fileId, err.message);
```

Comment:
Can you give me example of `err.message` value in this scenario?  Error log `File ${f} not found.` already explains what happened so I wonder how much more useful it gets.

Reply:
`err.message` contains the actual axios err message which may be useful to have in the object for debugging purposes. The `File ${f} not found.` will be displayed on the client rather than a complicated and unnecessarily detailed message

Reply:
This also explains the purpose of `extraData`

Reply:
Can you please give me an example for message from Axios? I'd love to see how it does look like.

Reply:
![image](https://user-images.githubusercontent.com/54070318/78568308-d3bb3b80-7819-11ea-9fe4-4258b2c973cc.png)